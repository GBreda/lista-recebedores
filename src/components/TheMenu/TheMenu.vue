<template>
  <div class="the-menu">
    <router-link
      class="the-menu__link"
      data-test-id="menu-link"
      to="/"
      active-class="the-menu__active"
      >Seus favorecidos</router-link
    >
  </div>
</template>

<style lang="scss" scoped>
.the-menu {
  background-color: $green;
  padding: 1.7rem 0 1.7rem 5rem;

  &__link {
    color: $white;
    padding-bottom: 1.2rem;
    text-decoration: none;
  }

  &__active {
    border-bottom: 0.3rem solid $white;
  }
}
</style>
