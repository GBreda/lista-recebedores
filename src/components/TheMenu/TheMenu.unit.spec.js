import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import { createRouter, createMemoryHistory } from 'vue-router'
import TheMenu from './TheMenu.vue'
import { nextTick } from 'vue'

const mockRouter = createRouter({
  history: createMemoryHistory(),
  routes: [
    {
      name: 'receivers-list',
      path: '/'
    }
  ]
})

const mountTheMenu = () => {
  return mount(TheMenu, {
    global: {
      plugins: [mockRouter]
    }
  })
}

describe('TheMenu', () => {
  let wrapper

  it('should mount the component', () => {
    wrapper = mountTheMenu()

    expect(wrapper.vm).toBeDefined()
  })

  it('should match the snapshot', () => {
    wrapper = mountTheMenu()

    expect(wrapper.element).toMatchSnapshot()
  })

  it('should push to the home route', async () => {
    await mockRouter.push('/')
    await mockRouter.isReady()

    wrapper = mountTheMenu()

    await wrapper.find("[data-test-id='menu-link']").trigger('click')
    await nextTick()

    expect(mockRouter.currentRoute.value.name).toBe('receivers-list')
  })
})
