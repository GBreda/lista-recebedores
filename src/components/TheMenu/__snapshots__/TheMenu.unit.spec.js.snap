// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TheMenu > should match the snapshot 1`] = `
<div
  class="the-menu"
  data-v-677364ff=""
>
  <a
    class="the-menu__link"
    data-test-id="menu-link"
    data-v-677364ff=""
    href="/"
  >
    Seus favorecidos
  </a>
</div>
`;
